src/python/pants/backend/awslambda/python/rules.py

```diff
@@ -54,6 +55,14 @@ class PythonAwsLambdaFieldSet(PackageFieldSet):
 async def package_python_awslambda(
     field_set: PythonAwsLambdaFieldSet, lambdex: Lambdex, union_membership: UnionMembership
 ) -> BuiltPackage:
+
+    if sys.platform == "darwin":
+        logger.warning(
+            "Building lambdas on macOS is not recommended. Lambdas must be built for Linux "
+            "and it is difficult to ensure that all transitive requirements have "
+            "prebuilt linux wheels."
```

Comment:
It's okay to run from macOS, it can often work perfectly well! The only risk is you will sometimes fail to build, particularly if your lambda uses any source distributions (sdists).

How about something like this?

> Building an AWS Lambda from macOS, which may fail to build. If your lambda uses any third-party dependencies without binary wheels (bdist) for Linux available, it will fail to build. If this happens, you will either need to update your dependencies to only use dependencies with pre-built wheels, or find a Linux environment to run `./pants package`. (See https://realpython.com/python-wheels/ for more about wheels.)
>
> (If the build does not raise an exception, it's safe to use macOS.)

(You can use `\n\n` for the paragraph break)